app/models/grouping.rb

```diff
@@ -754,6 +754,19 @@ def get_random_incomplete(current_role)
     results.where.not('groupings.id': self.id).order('RANDOM()').first&.grouping
   end
 
+  # Checks if a grouping uploaded any files
+  def has_submitted_files?
+    access_repo do |repo|
+      revision = repo.get_revision_by_timestamp(Time.current)
+
+      files = revision.tree_at_path('/', with_attrs: false).select do |_, obj|
```

Comment:
Use `assignment.repository_folder` instead of `'/'` (e.g., see `assignments_controller.rb` line 802). This will check for this grouping's folder for this specific assignment.